go-controller/pkg/ovn/ovn.go

```diff
@@ -268,6 +269,26 @@ func (oc *Controller) WatchNodes() error {
 			if err != nil {
 				logrus.Errorf("error creating Node Management Port for node %s: %v", node.Name, err)
 			}
+
+			mode, ok := node.Annotations[OvnNodeGatewayMode]
+			if !ok {
+				logrus.Errorf("gateway mode annotation not found for node %q ", node.Name)
+			}
+
+			if mode != string(config.GatewayModeDisabled) {
+				subnet, err := parseNodeHostSubnet(node)
+				if err != nil {
+					if err = oc.syncGatewayLogicalNetwork(node, mode); err != nil {
```

Comment:
L280: Did you mean `err == nil` here?

Reply:
@girishmg yeah I brought that up during internal review this morning and he's working on it